tests/integration/components/listbox-test.js

```diff
@@ -0,0 +1,3354 @@
+
+      // Verify that the first non-disabled listbox option is active
+      assertActiveListboxOption(options[2]);
+    });
+
+    test('should have no active listbox option upon Enter key press, when there are no non-disabled listbox options', async () => {
+      await render(hbs`
+        <Listbox as |listbox|>
+           <listbox.Button data-test="headlessui-listbox-button-1">Trigger</listbox.Button>
+           <listbox.Options data-test="headlessui-listbox-options-1" as |options|>
+             <options.Option @disabled={{true}} @value="a">
+               Option A
+             </options.Option>
+             <options.Option @disabled={{true}} @value="b">
+               Option B
+             </options.Option>
+             <options.Option @disabled={{true}} @value="c">
+               Option C
+             </options.Option>
+           </listbox.Options>
+         </Listbox>
+      `);
+
+      assertListboxButton({
+        state: ListboxState.InvisibleUnmounted,
+        attributes: { 'data-test': 'headlessui-listbox-button-1' },
+      });
+      assertListbox({ state: ListboxState.InvisibleUnmounted });
+
+      // Focus the button
+      getListboxButton().focus();
+
+      // Open listbox
+      await triggerKeyEvent(document.activeElement, 'keypress', 'Enter');
+
+      assertNoActiveListboxOption();
+    });
+
+    test('should be possible to close the listbox with Enter when there is no active listboxoption', async () => {
+      await render(hbs`
+        <Listbox as |listbox|>
+           <listbox.Button data-test="headlessui-listbox-button-1">Trigger</listbox.Button>
+           <listbox.Options data-test="headlessui-listbox-options-1" as |options|>
+             <options.Option @value="a">
+               Option A
+             </options.Option>
+             <options.Option @value="b">
+               Option B
+             </options.Option>
+             <options.Option @value="c">
+               Option C
+             </options.Option>
+           </listbox.Options>
+         </Listbox>
+      `);
+
+      assertListboxButton({
+        state: ListboxState.InvisibleUnmounted,
+        attributes: { 'data-test': 'headlessui-listbox-button-1' },
+      });
+      assertListbox({ state: ListboxState.InvisibleUnmounted });
+
+      // Open listbox
+      await click(getListboxButton());
+
+      // Verify it is visible
+      assertListboxButton({ state: ListboxState.Visible });
+
+      // Close listbox
+      await triggerKeyEvent(document.activeElement, 'keypress', 'Enter');
+
+      // Verify it is closed
+      assertListboxButton({ state: ListboxState.InvisibleUnmounted });
+      assertListbox({ state: ListboxState.InvisibleUnmounted });
+
+      // Verify the button is focused again
+      await assertActiveElement(getListboxButton());
+    });
+
+    test('should be possible to close the listbox with Enter and choose the active listbox option', async function () {
+      let handleChange = sinon.spy();
```

Comment:
We already have `testdouble` installed for creating spys and asserting their use; I'd rather we stick to one dependency of that variety than using two different ones in different test modules

Reply:
https://github.com/GavinJoyce/ember-headlessui/pull/76/commits/dac162c4e538f6f6bac6648751dcfbbd5949d242 has removed `ember-sinon` as a dependency.

I couldn't find where we use `testdouble` so I just implemented the assertions natively.

Reply:
Whoops, that's my bad! I added it in the `Transition` PR thats' been open a while and forgot this hasn't actually landed yet

https://github.com/GavinJoyce/ember-headlessui/pull/68/commits/0a5fefd5b2726d16106bbe45c7e1175453980f41

We could use `ember-sinon` instead of you have a hard preference there, or refactor the tests down the line to use `testdouble` once all of these PRs are merged